connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceTask.java

```diff
@@ -106,13 +107,28 @@ public void start(Map<String, String> props) {
         Map<TopicPartition, Long> topicPartitionOffsets = loadOffsets(taskTopicPartitions);
         consumer.assign(topicPartitionOffsets.keySet());
         log.info("Starting with {} previously uncommitted partitions.", topicPartitionOffsets.entrySet().stream()
-            .filter(x -> x.getValue() == 0L).count());
-        log.trace("Seeking offsets: {}", topicPartitionOffsets);
-        topicPartitionOffsets.forEach(consumer::seek);
+            .filter(x -> x.getValue() == NON_EXISTING_OFFSET_VALUE).count());
+
+        log.trace("Seeking offsets: {}", topicPartitionOffsets.entrySet().stream()
+                .filter(topicPartitionOffset ->
+                        topicPartitionOffset.getValue() != NON_EXISTING_OFFSET_VALUE));
+
+        topicPartitionOffsets.forEach(this::maybeSeek);
         log.info("{} replicating {} topic-partitions {}->{}: {}.", Thread.currentThread().getName(),
             taskTopicPartitions.size(), sourceClusterAlias, config.targetClusterAlias(), taskTopicPartitions);
     }
 
+    private void maybeSeek(TopicPartition topicPartition, Long offset) {
+        // Do not call seek on partitions that don't have an existing offset committed.
+        if (offset == NON_EXISTING_OFFSET_VALUE) {
```

Comment:
If we use `isUncommitted`:
```suggestion
        if (isUncommitted(offset)) {
```